addons/binding/org.openhab.binding.plclogo/src/main/java/org/openhab/binding/plclogo/handler/PLCDigitalHandler.java

```diff
@@ -0,0 +1,325 @@
+            value = new DecimalType(type == OpenClosedType.CLOSED ? 1 : 0);
+        }
+
+        final Channel channel = thing.getChannel(channelUID.getId());
+        Objects.requireNonNull(channel, "PLCDigitalHandler: Invalid channel found.");
+
+        setOldValue(getBlockFromChannel(channel), value);
+    }
+
+    @Override
+    protected void updateConfiguration(Configuration configuration) {
+        super.updateConfiguration(configuration);
+        synchronized (config) {
+            config = getConfigAs(PLCDigitalConfiguration.class);
+        }
+    }
+
+    @Override
+    protected boolean isValid(final String name) {
+        if (2 <= name.length() && (name.length() <= 4)) {
+            final String kind = getBlockKind();
+            if (Character.isDigit(name.charAt(1)) || Character.isDigit(name.charAt(2))) {
+                boolean valid = kind.equalsIgnoreCase("I") || kind.equalsIgnoreCase("NI");
+                valid = valid || kind.equalsIgnoreCase("Q") || kind.equalsIgnoreCase("NQ");
+                return name.startsWith(kind) && (valid || kind.equalsIgnoreCase("M"));
+            }
+        }
+        return false;
+    }
+
+    @Override
+    protected String getBlockKind() {
+        return config.getBlockKind();
+    }
+
+    @Override
+    protected int getNumberOfChannels() {
+        final String kind = getBlockKind();
+        final String family = getLogoFamily();
+        logger.debug("Get block number of {} LOGO! for {} blocks.", family, kind);
+
+        final Integer number = LOGO_BLOCK_NUMBER.get(family).get(kind);
+        return number == null ? 0 : number.intValue();
+    }
+
+    @Override
+    protected int getAddress(final String name) {
+        int address = super.getAddress(name);
+        if (address != INVALID) {
+            address = getBase(name) + (address - 1) / 8;
+        } else {
+            logger.warn("Wrong configurated LOGO! block {} found.", name);
+        }
+        return address;
+    }
+
+    @Override
+    protected void doInitialization() {
+        final Thing thing = getThing();
+        Objects.requireNonNull(thing, "PLCDigitalHandler: Thing may not be null.");
+
+        final Bridge bridge = getBridge();
+        Objects.requireNonNull(bridge, "PLCMemoryHandler: Bridge may not be null.");
+
+        logger.debug("Initialize LOGO! digital input blocks handler.");
+
+        synchronized (config) {
+            config = getConfigAs(PLCDigitalConfiguration.class);
+        }
+
+        super.doInitialization();
+        if (ThingStatus.OFFLINE != thing.getStatus()) {
+            final String kind = getBlockKind();
+            final String text = kind.equalsIgnoreCase("I") || kind.equalsIgnoreCase("NI") ? "input" : "output";
+
+            ThingBuilder tBuilder = editThing();
+
+            final String label = bridge.getLabel() == null ? "Siemens Logo!" : bridge.getLabel();
+            tBuilder.withLabel(label + ": digital " + text + "s");
```

Comment:
@dstivanov is this the (same) problem you mention in https://github.com/openhab/static-code-analysis/issues/198

@falkena what are you doing here, with which intention?

Reply:
@martinvw No, this is not the same. The Static Analysis Tooling checks that the `ThingHandler`does not use the `setLabel` method. Here you are using the `ThingBuilder` , which is used for construction of things and I think that this `withLabel` here is expected. Or am I overlooking something?